### PR TITLE
Add dump command

### PR DESCRIPTION
**Pull request by leplatrem**
- [x] Basic working code
- [x] subcommand `kinto-wizard dump`
- [x] Integration tests
- [x] Add load command
- [x] Make round-trip test pass :)

### CODE REVIEW

File: tests/test_functional.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        self.auth = os.getenv("AUTH", "user:pass")
+        self.file = os.getenv("FILE", "tests/kinto.yaml")
+        requests.post(self.server + "/__flush__")
+
+    def test_round_trip(self):
+        cmd = 'kinto-wizard {} --server={} --auth={}'
+
+        load_cmd = cmd.format("load {}".format(self.file),
+                              self.server, self.auth)
+        subprocess.check_call(load_cmd.split(" "))
```
Comment (Natim) [inline tests/test_functional.py]: Should we use a bash script instead?

File: .travis.yml [hunk 2]
```diff
@@ -7,6 +7,7 @@ env:
     - TOX_ENV=flake8
 install:
     - make install
```
Comment (Natim) [inline .travis.yml]: make install-dev?

File: kinto_wizard/kinto2yaml.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    return {
+        'permissions': permissions,
+        'collections': {
+            collection['id']: introspect_collection(client, bid, collection['id'])
+            for collection in collections
+        },
+        'groups': {
+            group['id']: introspect_group(client, bid, group['id'])
+            for group in groups
+        }
```
Comment (glasserc) [inline kinto_wizard/kinto2yaml.py]: Should we add `data` so that we can roundtrip?

File: kinto_wizard/yaml2kinto.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+                batch.create_bucket(bucket_id,
+                                    data=bucket_data,
+                                    permissions=bucket_permissions)
+            else:
+                bucket_current_groups = current_server_status[bucket_id]['groups']
+                bucket_current_collections = current_server_status[bucket_id]['collections']
+
+                # Patch the bucket if mandatory
+                current_bucket = current_server_status[bucket_id]
+                current_bucket_data = current_bucket.get('data', {})
```
Comment (glasserc) [inline kinto_wizard/yaml2kinto.py]: As noted above, buckets returned by `introspect_server` do not have `data`.

File: kinto_wizard/yaml2kinto.py [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+
+                if (current_bucket_data != bucket_data or
+                        current_bucket_permissions != bucket_permissions):
+                    batch.patch_bucket(bucket_id,
+                                       data=bucket_data,
+                                       permissions=bucket_permissions)
+
+            # 1.1 For each group, patch it if needed
+            for group_id, group_info in bucket_groups.items():
+                group_exists = bucket_exists and group_id in bucket_current_groups
```
Comment (glasserc) [inline kinto_wizard/yaml2kinto.py]: You don't really need to check `bucket_exists` here (although I guess it doesn't hurt).

File: kinto_wizard/kinto2yaml.py [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    }
+
+
+def introspect_collection(client, bid, cid):
+    logger.info("Fetch information of collection {!r}/{!r}".format(bid, cid))
+    collection = client.get_collection(bucket=bid, collection=cid)
+    return {
+        'permissions': collection['permissions'],
```
Comment (glasserc) [inline kinto_wizard/kinto2yaml.py]: Should we add `data` here too?

File: tests/test_functional.py (hunk 1 above)
Comment (glasserc) [inline tests/test_functional.py]: Or just call the code directly?

File: tests/test_functional.py [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
+        load_cmd = cmd.format("dump", self.server, self.auth)
+
+        with tempfile.NamedTemporaryFile() as fp:
+            # Dump content.
+            subprocess.check_call(load_cmd.split(" "), stdout=fp, stderr=open(os.devnull))
+
+            # Check that identical to original file.
+            diff_cmd = 'diff {} {}'.format(self.file, fp.name)
```
Comment (glasserc) [inline tests/test_functional.py]: We could also read the files ourselves and do `assertEqual`.

**Pull request merged**